src/components/searchBar/index.tsx

```diff
@@ -1,65 +1,168 @@
+    };
   }, []);
 
   return (
-    <form
-      className={cn(
-        'relative flex items-center',
+    <div ref={searchBarRef} className="w-[38rem]">
+      <div className={cn(
+        'flex items-center gap-4',
         classNames,
       )}
-    >
-      <div className="relative w-96">
-        <Icon
-          name="icon-search"
-          className="absolute left-3 top-1/2 -translate-y-1/2 "
-        />
-        <input
-          type="text"
-          name="search"
-          placeholder={label}
-          value={searchInput}
-          onChange={handleSearchInputChange}
+      >
+        <div
           className={cn(
-            'w-full p-3 pl-10',
-            'font-geist font-body2-regular',
-            'rounded border-b border-gray-300',
-            'bg-transparent dark:border-dev-purple-700',
-            'placeholder:font-geist placeholder:font-body2-regular dark:placeholder-dev-purple-300',
+            'flex w-full items-center gap-1 p-3',
+            'border-b border-gray-300',
+            'dark:border-dev-purple-700 dark:bg-transparent',
+            'duration-300 ease-out hover:border-dev-pink-500',
           )}
-        />
+        >
+          <Icon name="icon-search" />
+          <input
+            type="text"
+            placeholder={label}
+            value={searchInput}
+            onChange={handleSearchInputChange}
+            className={cn(
+              'w-full dark:bg-transparent',
+              'caret-dev-pink-500 focus-visible:outline-none',
+              'placeholder:font-geist placeholder:font-body2-regular dark:placeholder-dev-purple-300',
+            )}
+          />
+        </div>
+        <button
+          className={cn(
+            'px-6 py-2',
+            'bg-dev-purple-700 text-dev-purple-300',
+            'transition-all duration-300 hover:bg-dev-purple-900',
+            'dark:bg-dev-purple-50 dark:text-dev-black-1000 dark:hover:bg-dev-purple-200',
+          )}
+        >
+          <span className="font-geist font-body2-bold">Search</span>
+        </button>
       </div>
-      <button
-        disabled
-        className={cn(
-          'ml-4 px-6 py-2',
-          'cursor-pointer bg-dev-purple-700 text-dev-white-200',
-          'transition-all duration-300 hover:bg-dev-purple-900',
-          'dark:bg-dev-purple-50 dark:text-dev-black-1000 dark:hover:bg-dev-purple-200',
-        )}
-      >
-        <span className="font-geist font-body2-bold">
-          Search
-        </span>
-      </button>
-    </form>
+      {isResultsVisible && (
+        <Results
+          results={results}
+          handleOpenModal={handleOpenModal}
+          type={type}
+          classNames={cn(
+            'opacity-0 transition-all',
+            {
+              'opacity-100 translate-y-0 pointer-events-auto': searchInput,
+            })
+          }
+        />
+      )}
+
+      {
+        refSelectedExtrinsic.current && (
```

Comment:
Looks a bit inconsistent to use both `condintion && <Component>` and `condition ? <Component> : null` across the codebase, could we consider using just one of the patterns?

Reply:
I agree that I will use the condition && with <Component>.